### PR TITLE
Miscellaneous improvements regarding network splitting and simplification

### PR DESCRIPTION
**Pull request by MaLoefUDS**
### Prerequisites

- [x] I adhere to the coding conventions (described [here](../CONTRIBUTING.md)) in my code.
- [x] I have updated the copyright headers of the files I have modified.
- [x] I have written appropriate commit messages, i.e., I have recorded the goal, the need, the needed changes, and the location of my code modifications for each commit. This includes also, e.g., referencing to relevant issues.
- [x] I have put signed-off tags in *all* commits.
- [ ] I have updated the changelog file [NEWS.md](../NEWS.md) appropriately.
- [ ] I have checked whether I need to adjust the showcase file `showcase.R` with respect to my changes.
- [x] The pull request is opened against the branch `dev`.

### CODE REVIEW

File: util-split.R [hunk 1]
```diff
@@ -1016,6 +1016,8 @@ split.data.by.time.or.bins = function(project.data, splitting.length, bins, spli
                         findInterval(df[["date"]], bins.date, all.inside = FALSE)
             ## split data according to df.bins
             df.split = split(df, df.bins)
+            ## remove data before the first bin
+            df.split = df.split[names(df.split) != "0"]
```
Comment (bockthom) [inline util-split.R:1020]: Two questions: "0" as a character looks weird – although I know that it is correct. But could we add a comment at the end of the line that explains why we are looking for the "0" as character?

And do we have tests for this edge case? Not sure whether and how we can test this, but maybe by triggering a similar situation as the one in which we have found this bug?

File: util-networks.R [hunk 2]
```diff
@@ -1993,6 +1993,43 @@ delete.isolates = function(network) {
     return(network.no.isolates)
 }
 
+#' Remove duplicate edges from the given network.
```
Comment (bockthom) [inline util-networks.R:1996]: Maybe we should add an additional sentence describing what "duplicate edges" mean.
It may also be useful to add a sentence in the readme, when we introduce examples to explain temporal order and directedness, that duplicate edges in a particular case can be removed by calling this function. Otherwise, nobody will use this function because nobody is aware of it.

Let's also check if there is a comment on duplicate edges in one of the edge creation functions. If so, we could refer to this function in the comment. If not, then it is not really necessary.

File: tests/test-networks.R [hunk 3]
```diff
@@ -317,6 +317,47 @@ test_that("Remove isolated authors given a specific edge type", {
 
 })
 
+test_that("Remove duplicate edges", {
+
+    ## configurations
+    proj.conf = ProjectConf$new(CF.DATA, CF.SELECTION.PROCESS, CASESTUDY, ARTIFACT)
+    net.conf = NetworkConf$new()
+    net.conf$update.value("author.respect.temporal.order", TRUE)
```
Comment (bockthom) [inline tests/test-networks.R:325]: For better comprehensibility of this test, it may be useful to also state which value of `author.directed` is used. Because this should make a difference on whether edges are actually removed or not, if I am not mistaken.

**Review by bockthom — COMMENTED** [review]
Thanks! This PR looks good to me. I just have a few comments regarding documentation and tests.

File: util-split.R (hunk 1 above)
Comment (MaLoefUDS) [reply util-split.R:1020]: About the comment, yes. But your second point really is not testable. This issue does not really propagate to anything visible from the outside, it’s just internal. The reason for this is that ranges with names that are not corresponding to bin labels are dropped anyways just a few lines further down.

File: README.md [hunk 4]
```diff
@@ -318,7 +318,7 @@ Based on the above raw data, we get the following author networks with relation
   </tr>
 </table>
 
-When constructing author networks with respecting the temporal order, there is one edge for each answer in a mail thread from the answer's author to the senders of every previous e-mail in this mail thread. Note that this can lead to duplicated edges if an author has sent several previous e-mails to the mail thread (see the duplicated edges `A –(3)– B` in the above example). This also leads to loop edges if an author of an answer has already sent an e-mail to this thread before (see the edge `A –(2)– A`).
+When constructing author networks with respecting the temporal order, there is one edge for each answer in a mail thread from the answer's author to the senders of every previous e-mail in this mail thread. Note that this can lead to duplicated edges if an author has sent several previous e-mails to the mail thread (see the duplicated edges `A –(3)– B` in the above example). These may be disambiguated again using the `remove.duplicate.edges` function. Furthermore, respecting the temporal order also leads to loop edges if an author of an answer has already sent an e-mail to this thread before (see the edge `A –(2)– A`).
```
Comment (bockthom) [inline README.md:321]: I don't like the word "disambiguated" here as does not clearly indicate what actually happens.
What about "conflated" ?

Everything else looks fine to me.

**Review by bockthom — CHANGES_REQUESTED** [review]

**Comment by bockthom** [comment]
Just one final question before I can approve this PR: The splitting fix regarding the elements before the first bin does not have any user-visible effects? If this really is the case, I am fine with not mentioning it. If it can have user-visible effects, we should mention it in the NEWS.md.

**Comment by MaLoefUDS** [comment]
Yup, it has no user-visible effect. That is why I found it hard to justify mentioning it in the NEWS and decided against it 😄

**Review by bockthom — APPROVED** [review]
